shared/src/chain_state.rs

```diff
@@ -455,42 +489,42 @@ impl<CS: ChainStore> ChainState<CS> {
         for entry in tx_pool.gap.entries() {
             if self.contains_proposal_id(entry.key()) {
                 let entry = entry.remove();
-                entries.push((entry.cycles, entry.transaction));
+                entries.push((entry.cycles, entry.size, entry.transaction));
             }
         }
 
         // try move pending to proposed
         for entry in tx_pool.pending.entries() {
             if self.contains_proposal_id(entry.key()) {
                 let entry = entry.remove();
-                entries.push((entry.cycles, entry.transaction));
+                entries.push((entry.cycles, entry.size, entry.transaction));
             } else if self.contains_gap(entry.key()) {
                 let entry = entry.remove();
-                gaps.push((entry.cycles, entry.transaction));
+                gaps.push((entry.cycles, entry.size, entry.transaction));
             }
         }
 
         // try move conflict to proposed
         for entry in tx_pool.conflict.entries() {
             if self.contains_proposal_id(entry.key()) {
                 let entry = entry.remove();
-                entries.push((entry.cycles, entry.transaction));
+                entries.push((entry.cycles, entry.size, entry.transaction));
             } else if self.contains_gap(entry.key()) {
                 let entry = entry.remove();
-                gaps.push((entry.cycles, entry.transaction));
+                gaps.push((entry.cycles, entry.size, entry.transaction));
             }
         }
 
-        for (cycles, tx) in entries {
+        for (cycles, size, tx) in entries {
             let tx_hash = tx.hash().to_owned();
-            if let Err(e) = self.proposed_tx_and_descendants(&mut tx_pool, cycles, tx) {
+            if let Err(e) = self.proposed_tx_and_descendants(&mut tx_pool, cycles, size, tx) {
                 debug!(target: "tx_pool", "Failed to add proposed tx {:x}, reason: {:?}", tx_hash, e);
```

Comment:
Missing the decrease of pool size here? Entries are removed but pool size is not change

Reply:
see  `proposed_tx`